drizzlepac/hlautils/poller_utils.py

```diff
@@ -0,0 +1,315 @@
+                    if filename[1][10:13].lower().endswith("flt"):
+                        filetype = "drz"
+                    prev_det_indx = det_indx
+
+                # Generate the full product information string: 
+                # proposal_id, obset_id, instrument, detector, ipppssoot, filter, and filetype
+                prod_info = (filename[0] + " " + filetype).lower()
+
+                # Set up the single exposure product dictionary
+                sep = SEP_STR.format(sep_indx)
+                obset_products[sep] = {'info': prod_info,
+                                       'files': [filename[1]]}
+
+                # Create a single exposure product object
+                prod_list = prod_info.split(" ")
+                sep_obj = ExposureProduct(prod_list[0], prod_list[1], prod_list[2], prod_list[3], filename[1], prod_list[5], prod_list[6])
+                sep_list.append(sep_obj)
+
+                # Set up the filter product dictionary and create a filter product object
+                # Initialize `info` key for this filter product
+                if not obset_products[fprod]['info']:
+                    obset_products[fprod]['info'] = prod_info
+                    # Create a filter product object for this instrument/detector
+                    filt_obj = FilterProduct(prod_list[0], prod_list[1], prod_list[2], prod_list[3], prod_list[4], prod_list[5])
+                # Append exposure object to the list of exposure objects for this specific filter product
+                filt_obj.add_member(sep_obj)
+                # Populate filter product with input filename
+                obset_products[fprod]['files'].append(filename[1])
+
+                # Set up the total detection product dictionary and create a total detection product object
+                # Initialize `info` key for total detection product
+                if not obset_products[totprod]['info']:
+                    obset_products[totprod]['info'] = prod_info
+                    # Create a total detection product object for this instrument/detector
+                    tdp_obj = TotalProduct(prod_list[0], prod_list[1], prod_list[2], prod_list[3], prod_list[4])
+                # Append exposure object to the list of exposure objects for this specific total detection product
+                tdp_obj.add_member(sep_obj)
+                # Populate total detection product with input filename
+                obset_products[totprod]['files'].append(filename[1])
+
+                # Increment single exposure index
+                sep_indx += 1
+
+            # Append filter object to the list of filter objects for this specific total detection product
+            tdp_obj.add_product(filt_obj)
+
+            # Add the FilterProduct to the list of FilterProducts
+            filt_list.append(filt_obj)
+
+        # Add the TotalProduct to the list of TotalProducts
+        tdp_list.append(tdp_obj)
+
+    # Just for debugging.  
```

Comment:
Debugging statements should either be removed or set to 'log.debug' prior to merging.

Reply:
Removed debug cruft.